src/v2/Apps/Order/Components/SavedAddresses.tsx

```diff
@@ -171,20 +171,36 @@ const SavedAddresses: React.FC<SavedAddressesProps> = props => {
 
   const addAddressButton = (
     <>
-      <Button
-        mt={addressList.length > 0 ? 3 : 0}
-        variant="primaryBlack"
-        size="large"
-        onClick={() => {
-          setShowAddressModal(true),
-            setModalDetails({
-              addressModalTitle: "Add new address",
-              addressModalAction: "createUserAddress",
-            })
-        }}
-      >
-        Add new address
-      </Button>
+      {addressList.length > 0 ? (
+        <Button
+          variant="secondaryOutline"
+          width={159}
+          onClick={() => {
+            setShowAddressModal(true),
+              setModalDetails({
+                addressModalTitle: "Add address",
+                addressModalAction: "createUserAddress",
+              })
+          }}
+        >
+          Add a new address
+        </Button>
+      ) : (
+        <Button
```

Comment:
Can you describe this use-case more? I don't see it in the designs (but could be looking in the wrong place!).

Reply:
As I understand, if there are no saved addresses, one primaryBlack button is shown. 
If there is at least one saved address then secondaryOutline button is shown. But primaryBlack can be shown only on Profile settings ...
I can't check this on localy Force... maybe I need to configure .env file

![image](https://user-images.githubusercontent.com/55637696/121565345-0d429480-ca25-11eb-8a0d-285f8791d763.png)

Reply:
Oh interesting, so this component is re-used in the profile view?

My first thought is it might be fine to keep it as a white button instead of updating to be black. If we do want to keep this component in-use in both places, then perhaps we should pass a prop from the parent view to determine button color (as it seems the logic is more reflective of that vs. how many addresses there are).

@sepans since I think you may have worked on saved addresses? -- how does this sound?

Reply:
SavedAddresses component (which contains "Add new address" button) is re-used in the profile view.

Reply:
Yes this component is being reused in `UserSettingsAddresses`. great catch @Serge0n 💯 

@sweir27 if you mean having like (psydocoding...)
```tsx
const buttonVariant = whoIsMyParent === '....' && hasSavedAddress ? ... : ....
const addButtonProps = { variant: buttonVariant ...}
<Button {...addButtonProps}>...
```
It sounds good to me. Also wondering if we want to use this opportunity to make the address edit functionality under payment and addresses settings and checkout flow as identical as possible.

Reply:
@sepans Yes! And I agree with making the behavior the same eventually 😅.

Not for this PR, but we may want to move this shared component out of `Order` since it's a little confusing to have it there but also be shared!